components/builder-core/src/access_token.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::PathBuf;
+use chrono::Utc;
+
+use hab_net::privilege::FeatureFlags;
+use error::{Error, Result};
+use integrations::{encrypt, decrypt};
+use protocol::{message, sessionsrv};
+
+pub const BUILDER_ACCOUNT_ID: u64 = 0;
+const ACCESS_TOKEN_PREFIX: &'static str = "TOKN";
+
+const BUILDER_TOKEN_EXPIRATION_SECONDS: i64 = 7200; // 2 hours
+const USER_TOKEN_EXPIRATION_SECONDS: i64 = 315360000; // 10 years
+
+pub fn generate_bldr_token(key_dir: &PathBuf) -> Result<String> {
+    generate_access_token(
+        key_dir,
+        BUILDER_ACCOUNT_ID,
+        FeatureFlags::all().bits(),
+        BUILDER_TOKEN_EXPIRATION_SECONDS,
+    )
+}
+
+pub fn generate_user_token(key_dir: &PathBuf, account_id: u64, privileges: u32) -> Result<String> {
+    generate_access_token(
+        key_dir,
+        account_id,
+        privileges,
+        USER_TOKEN_EXPIRATION_SECONDS,
```

Comment:
Similar comment as above, but another nicety of using `Duration` is that it allows you to write `chrono::Duration:: max_value()`, assuming there's nothing particularly special about 10 years.

Reply:
Changing from 10 years to Duration::max_value() increases generated token size by 4 chars .. go figure.  Code is cleaner though, so I'll just go with it.